### PR TITLE
chore: merging DipperCL into master

### PR DESCRIPTION
**Pull request by Charles546**
#### Description

This is to merge DipperCL branch into master and preparing to release 1.0.0

**DO NOT SQUASH**

### CODE REVIEW

File: build/docker/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.12.9-alpine AS build
+FROM golang:1.13.1-alpine AS build
```
Comment (wyardley) [inline build/docker/Dockerfile]: CI updates should be in this commit too

File: docs/howtos/setup_local.md [hunk 2]
```diff
@@ -97,13 +84,22 @@ git add init.yaml
 git commit -m 'init' -a
 ```
 
- * Start your daemon with the local root repo
+## Start honeydipper daemon
```
Comment (wyardley) [inline docs/howtos/setup_local.md]: capital Honeydipper?

File: docs/interpolation.md [hunk 3]
```diff
@@ -105,15 +102,55 @@ For example:
...
+
+```yaml
+workflows:
+  find_first:
+    call_workflow: show_name
+    with:
+      name: $ctx.name,ctx.full_name,ctx.nick_name # choose the first non empty value from the listed varialbes
+```
+
+We can also specify a default value with quotes either single quote, double quote or back tick, if all the listed variables
```
Comment (wyardley) [inline docs/interpolation.md]: "with quotes, either single quotes, double quotes, or backticks,"

File: docs/interpolation.md [hunk 4]
```diff
@@ -105,15 +102,55 @@ For example:
...
+      timeout: $ctx.timeout,ctx.default_timeout,"1800"
+      previous: $?ctx.previous
+```
+
+
 ## Inline go template
 
-Besides the `:path:` prefix, we can also use inline go template to access the workflow context data. The inline go template can be used in
+Besides the `$` prefix, we can also use inline go template to access the workflow context data. The inline go template can be used in
 workflow definitions(data, content), workflow conidtion, and function parameters.
```
Comment (wyardley) [inline docs/interpolation.md]: condition

File: docs/documenting.md [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+## Building
+
+In order to build the document for local viewing, follow below steps.
+
+ 1. install sphinx following the [sphinx installation document](http://www.sphinx-doc.org/en/master/usage/installation.html)
+ 2. install markdown extension for sphinx following the [recommonmark installation document](https://www.sphinx-doc.org/en/master/usage/markdown.html)
+ 3. install the `read the docs` theme for sphinx following the [readthedoc theme installation document](https://sphinx-rtd-theme.readthedocs.io/en/latest/installing.html)
+ 4. clone the `honeydipper-sphinx` repo
+    ```bash
+    git clone https://github.com/honeydipper/honeydipper-sphinx.git
```
[the remark this answers is not in the document]
Comment (Charles546) [reply docs/documenting.md]: The docs are still in that repo.

**Comment by Charles546** [comment]
Just ran the whole directory through a spell checker, we shouldn't have any spelling errors now. :)

**Pull request merged**